src/dbnode/persist/fs/merger_test.go

```diff
@@ -450,7 +451,12 @@ func testMergeWith(
 				})
 				return nil
 			},
-			Close: func() error { return nil },
+			DeferClose: func() (persist.DataCloser, error) {
+				return func() error {
+					deferClosed = true
```

Comment:
nit: worth adding `require.False(t, deferClose)` to ensure we don't double close?

Reply:
+1